logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java

```diff
@@ -81,10 +81,11 @@ public void sendEvents(final PipelineOutput sender,
                         if (lastResponse != null && lastResponse.getStatus() == PipelineInput.ReceiveStatus.FAIL) {
                             // when last call to internalReceive generated a fail, restart from the
                             // fail position to avoid reprocessing of some events in the downstream.
-                            lastFailedPosition = lastResponse.getSequencePosition();
+                            // we get 0~clones.length position on failure, keep tracking failed position to correctly copy from original orderedEvents
+                            lastFailedPosition += lastResponse.getSequencePosition();
 
                             logger.warn("Attempted to send event to '{}' but that address reached error condition. " +
-                                    "Will Retry. Root cause {}", address, lastResponse.getCauseMessage());
+                                    "Will Retry. Root cause {} Failed at {} position.", address, lastResponse.getCauseMessage(), lastFailedPosition);
```

Comment:
🤔 I think this message in general is a little misleading, and want to avoid making it more so. The failure occurred while sending a _batch_ of events to the output, and the `lastFailedPosition` is _really_ the marker for the last known _success_.

Maybe something like this indicating how many events are remaining?

~~~ java
                            logger.warn("Attempted to send events to '{}' but that address reached error condition with {} events remaining. " +
                                    "Will Retry. Root cause {}", address, orderedEvents.length - lastFailedPosition, lastResponse.getCauseMessage());
~~~